File: src/main/java/org/jukeboxmc/block/BlockPistonarmcollision.java
package org.jukeboxmc.block;

import org.jukeboxmc.block.direction.BlockFace;
import org.jukeboxmc.item.ItemPistonarmcollision;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockPistonarmcollision extends Block {

    public BlockPistonarmcollision() {
        super( "minecraft:pistonarmcollision" );
    }

    @Override
    public ItemPistonarmcollision toItem() {
        return new ItemPistonarmcollision();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.PISTONARMCOLLISION;
    }

    public void setBlockFace( BlockFace blockFace ) {
        this.setState( "facing_direction", blockFace.ordinal() );
    }

    public BlockFace getBlockFace() {
        return this.stateExists( "facing_direction" ) ? BlockFace.values()[this.getIntState( "facing_direction" )] : BlockFace.NORTH;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockPistonArmCollision.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.PistonArmCollision
import org.jukeboxmc.api.block.data.BlockFace
import org.jukeboxmc.server.block.JukeboxBlock

class BlockPistonArmCollision(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    PistonArmCollision {

    override fun getFacingDirection(): BlockFace {
        return BlockFace.entries[this.getIntState("facing_direction")]
    }

    override fun setFacingDirection(value: BlockFace): PistonArmCollision {
        return this.setState("facing_direction", value.ordinal)
    }
}